nixos/modules/hardware/printers.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, services, ... }:
+with lib;
+let
+  cfg = config.hardware.printers;
+  ensurePrinter = p: ''
+    ${pkgs.cups}/bin/lpadmin -p '${p.name}' -E ${optionalString (p.location != null) "-L '${p.location}'"} ${optionalString (p.description != null) "-D '${p.description}'"} -v '${p.deviceUri}' -m '${p.model}'
+  '';
+  ensureDefaultPrinter = name: ''
+    ${pkgs.cups}/bin/lpoptions -d '${name}'
+  '';
+
+  forbiddenChars  = str: filter (c: c == "#" || c == "/") (stringToCharacters str) == [];
+  # TODO: is it possible to combine the forbiddenChars with that regex?
+  # test by using `nix repl` and then `builtins.match "[[:graph:]]+" "teststring#"`
+  printerName = (types.addCheck (types.strMatching "[[:graph:]]+") forbiddenChars)
+    // { description = "printable string without spaces, # and /"; };
+
+
+in
+{
+  options = {
+    hardware.printers = {
+      ensureDefaultPrinter = mkOption {
+        type = types.nullOr printerName;
+        default = null;
+        description = ''
+          Ensures the named printer is the default CUPS printer / printer queue.
+        '';
+      };
+      ensurePrinters = mkOption {
+        description = ''
+          Will regularly ensure that the given CUPS printers are configured as declared here.
+          If a printer's options are manually changed afterwards, they will be overwritten eventually.
+          This option will never delete any printer, even if removed from this list.
+          You can check existing printers with <literal>lpstat -s</literal>
+          and remove printers with <literal>lpadmin -x \<printer-name\></literal>.
```

Comment:
The part with `\<printer-name\>` breaks eval for me with the following error:
```
intermediate.xml:3157: parser error : error parsing attribute name
and remove printers with <literal>lpadmin -x \<printer-name\></literal>.
                                                           ^
intermediate.xml:3157: parser error : attributes construct error
and remove printers with <literal>lpadmin -x \<printer-name\></literal>.
                                                           ^
intermediate.xml:3157: parser error : Couldn't find end of Start Tag printer-name line 3157
and remove printers with <literal>lpadmin -x \<printer-name\></literal>.
                                                           ^
unable to parse intermediate.xml
```

I was able to build the manual by removing `\<` and `\>` from the description.

Reply:
Seems like that is not how you would escape pointy brackets in docbook. Well, just removed them, people will figure out it's a placeholder.

Reply:
In any XML-based language, one should be able to use the entity references `&lt;` and `&gt;` to represent `<` and `>`.

Reply:
Could have guessed that, admittedly. Gauging between using the regular `<option>` syntax for required man page arguments and keeping to the lpadmin man page which does not use anything like that, I switched back to `<option>`.